packages/addon/lua/gelly/mods/blood-mod/init.lua

```diff
@@ -72,41 +71,80 @@ local BLOOD_WEAPON_CONFIGS = { -- IMPLEMENT SPECIFIC WEAPON DAMAGE
 	},
 }
 
-local function getConfig(damageType)
-	for _, config in ipairs(BLOOD_CONFIGS) do
+local BLOOD_COLOR_MATERIALS = {
+	[BLOOD_COLOR_RED] = {
+		Roughness = 0,
+		IsSpecularTransmission = true,
+		RefractiveIndex = 1.373,
+		Absorption = Vector(0.05, 0.5, 0.5),
+		DiffuseColor = Vector(0, 0, 0),
+	},
+	[BLOOD_COLOR_YELLOW] = {
+		Roughness = 0,
+		IsSpecularTransmission = true,
+		RefractiveIndex = 1.373,
+		Absorption = Vector(0.25, 0.3, 0.5),
+		DiffuseColor = Vector(0, 0, 0),
+	},
+	[BLOOD_COLOR_GREEN] = {
+		Roughness = 0,
+		IsSpecularTransmission = true,
+		RefractiveIndex = 1.373,
+		Absorption = Vector(0.3, 0.25, 0.5),
+		DiffuseColor = Vector(0, 0, 0),
+	},
+	[BLOOD_COLOR_ANTLION] = {
+		Roughness = 0,
+		IsSpecularTransmission = true,
+		RefractiveIndex = 1.373,
+		Absorption = Vector(0.4, 0.15, 0.5),
+		DiffuseColor = Vector(0, 0, 0),
+	},
+	[BLOOD_COLOR_ZOMBIE] = {
+		Roughness = 0,
+		IsSpecularTransmission = true,
+		RefractiveIndex = 1.373,
+		Absorption = Vector(0.2, 0.35, 0.5),
+		DiffuseColor = Vector(0, 0, 0),
+	},
+	[BLOOD_COLOR_ANTLION_WORKER] = {
+		Roughness = 0,
+		IsSpecularTransmission = true,
+		RefractiveIndex = 1.373,
+		Absorption = Vector(0.45, 0.1, 0.5),
+		DiffuseColor = Vector(0, 0, 0),
+	},
+}
+
+local function getDamageTypeConfig(damageType)
+	for _, config in ipairs(DAMAGE_TYPE_BLOOD_CONFIGS) do
         if bit.band(damageType, config.DamageFlags) ~= 0 then
             return config
         end
 	end
 	return nil
 end
 
-local function getWeaponConfig(weaponClassName)
-	for _, config in ipairs(BLOOD_WEAPON_CONFIGS) do
-		if weaponClassName == config.ClassName then
-			return config
-		end
-	end
-	return nil
-end
+local function getConfig(attacker, damageType)
+	local config = nil
 
-local function sprayBlood(damageType, attacker, position, force, damage)
-	local normal = Vector()
-	if attacker == LocalPlayer() then
-		normal = LocalPlayer():GetAimVector()
-	else
-		normal = force:GetNormalized()
+	if attacker:IsValid() and (attacker:IsNPC() or attacker:IsPlayer()) then
+		config = WEAPON_BLOOD_CONFIGS[attacker:GetActiveWeapon():GetClass()]
```

Comment:
nit: Could just `return` here (and in the if statement below)

This would also let you remove the local `config` variable

Reply:
But the if statement below requires the returned config from this one

Reply:
The if statement below only uses it because it has to check whether _this_ if statement was entered

If you made this return, that check below is redundant